components/style/values/computed/counters.rs

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Computed values for counter properties
+
+use std::fmt;
+use style_traits::{CssWriter, ToCss};
+use values::CustomIdent;
+use values::computed::{Context, ToComputedValue};
+use values::generics::counters::CounterIntegerList;
+use values::specified::{CounterIncrement as SpecifiedCounterIncrement, CounterReset as SpecifiedCounterReset, Integer};
+
+type ComputedIntegerList = CounterIntegerList<i32>;
+
+/// A computed value for the `counter-increment` property.
+#[derive(Clone, Debug, MallocSizeOf, PartialEq)]
+pub struct CounterIncrement(ComputedIntegerList);
+
+impl CounterIncrement {
+    /// Returns the `none` value.
+    #[inline]
+    pub fn none() -> CounterIncrement {
+        CounterIncrement(CounterIntegerList::new(Vec::new()))
+    }
+
+    /// Returns a new computed `counter-increment` object with the given values.
+    pub fn new(vec: Vec<(CustomIdent, i32)>) -> CounterIncrement {
+        CounterIncrement(ComputedIntegerList::new(vec))
+    }
+
+    /// Returns a clone of the values of the computed `counter-increment` object.
+    pub fn get_values(&self) -> &[(CustomIdent, i32)] {
+        self.0.get_values()
+    }
+}
+
+impl ToCss for CounterIncrement {
+    fn to_css<W>(&self, dest: &mut CssWriter<W>) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        self.0.to_css(dest)
+    }
+}
+
+impl ToComputedValue for SpecifiedCounterIncrement {
+    type ComputedValue = CounterIncrement;
+
+    fn to_computed_value(&self, context: &Context) -> Self::ComputedValue {
+        CounterIncrement::new(self.get_values().iter().map(|&(ref name, ref value)| {
```

Comment:
This doesn't need to do this, instead, let's use `self.0.to_computed_value()` and `derive(ToComputedValue)`.

Reply:
(I think `CounterIntegerList` already implements `ToComputedValue`, so you'd only need to call it here).